### ISSUE TITLE
Ability to write custom checks on provider block

### ISSUE THREAD

**Issue by KarthikVenkatraman** [bridgecrewio/checkov#2038]
While checkov allows us to write custom checks involving various resource types in terraform, am not sure if there is a way to add custom checks for the **provider block** in terraform. The default check that runs for the provider block does scan for secrets, however am looking to check if the provider block has a **default_tags** definition.. The reason I need to check this is that while we can write custom checks to check tags for all supported resource types, if we would need to avoid having to repeat the tags for individual resources and use the **default_tags** feature in terraform to apply a same set of tags to all resources, then there should be a way to check this and If defined then not check the individual resource types for the tags.

**Comment by gruebel** [comment]
@KarthikVenkatraman yeah it should be possible to write a custom checks for a provider. Just copy this file https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/provider/aws/credentials.py and of course don't forget the `__init__.py` like for other custom checks and you are all set.

**Comment by KarthikVenkatraman** [comment]
Hi @gruebel Thank you so much for the reply. Is it possible to write the check only with yaml rules and not having to do this in Python please ?

**Comment by gruebel** [comment]
as far as I know, the YAML checks only work with normal resources
